$OpenBSD: patch-tools_clang_include_clang_Basic_BuiltinsX86_def,v 1.1 2017/10/07 12:08:33 ajacoutot Exp $

[X86] Disable _mm512_maskz_set1_epi64 intrinsic on 32-bit targets to prevent a
backend isel failure.

The __builtin_ia32_pbroadcastq512_mem_mask we were previously trying to use in
32-bit mode is not implemented in the x86 backend and causes isel to fail in
release builds. In debug builds it fails even earlier during legalization with
an llvm_unreachable.

Index: tools/clang/include/clang/Basic/BuiltinsX86.def
--- tools/clang/include/clang/Basic/BuiltinsX86.def.orig
+++ tools/clang/include/clang/Basic/BuiltinsX86.def
@@ -976,7 +976,6 @@ TARGET_BUILTIN(__builtin_ia32_pmuludq512, "V8LLiV16iV1
 TARGET_BUILTIN(__builtin_ia32_ptestmd512, "UsV16iV16iUs", "", "avx512f")
 TARGET_BUILTIN(__builtin_ia32_ptestmq512, "UcV8LLiV8LLiUc", "", "avx512f")
 TARGET_BUILTIN(__builtin_ia32_pbroadcastd512_gpr_mask, "V16iiV16iUs", "", "avx512f")
-TARGET_BUILTIN(__builtin_ia32_pbroadcastq512_mem_mask, "V8LLiLLiV8LLiUc", "", "avx512f")
 TARGET_BUILTIN(__builtin_ia32_loaddqusi512_mask, "V16iiC*V16iUs", "", "avx512f")
 TARGET_BUILTIN(__builtin_ia32_loaddqudi512_mask, "V8LLiLLiC*V8LLiUc", "", "avx512f")
 TARGET_BUILTIN(__builtin_ia32_loadups512_mask, "V16ffC*V16fUs", "", "avx512f")
